tools/build-tools/src/repoPolicyCheck/handlers/lockfiles.ts

```diff
@@ -23,7 +22,8 @@ export const handlers: Handler[] = [
             if (matches !== null) {
                 const results: string[] = [];
                 const containsBadUrl = matches.some((value) => {
-                    if (value.startsWith(`https://registry.npmjs.org`)) {
+                    const url = new URL(value);
+                    if (url.hostname !== `registry.npmjs.org` || url.protocol !== `https:`) {
```

Comment:
is this supposed to reverse the logic?  it wasn't obvious to me what the problem was from the linked issue

Reply:
You're totally right. 🤦🏻 Fixed in the latest iteration.